architecture/adr-list/adr-001-did-resolver.md

```diff
@@ -0,0 +1,300 @@
+
+## Context
+
+[DID resolution](https://meet.google.com/xip-iwcs-kdj?authuser=0) is the process through which DIDs, associated DIDDocs, and other resources can be accessed using the "Read" functions of a DID method (in our case, the [cheqd DID method](https://docs.cheqd.io/node/architecture/adr-list/adr-002-cheqd-did-method)).
+
+There is [a defined algorithm with standardised behaviour for expected and unexpected inputs](https://w3c-ccg.github.io/did-resolution/#resolving-algorithm) sent to DID resolution software. The purpose of a DID Resolver for `did:cheqd` would be to implement such an algorithm for interactions with the cheqd ledger.
+
+### Resolving a `did:cheqd` entry
+
+All conforming DID resolvers implement `resolve` and `resolveRepresentation` abstract functions, as defined in the [DID Resolution specification](https://w3c-ccg.github.io/did-resolution/#resolving).
+
+#### Resolve function
+
+```js
+resolve(did, resolutionOptions) → 
+« didResolutionMetadata, didDocument, didDocumentMetadata »
+```
+
+The `resolve` function is intended to provide [resolution of a DID to as a JSON/JSON-LD map](https://w3c.github.io/did-core/#json) with the relevant DID Resolution Metadata, DID Document and DID Document Metadata populated.
+
+For example, `did:cheqd:testnet:DAzMqo4MDMxCjgwM` resolves to the following response, containing Resolution Metadata, DIDDoc, and DIDDoc Metadata:
+
+```jsonc
+{
+    "didResolutionMetadata": {
+        "contentType": "application/did+ld+json",
+        "retrieved": "2022-08-01T19:34:30Z",
+        "did":{
+            "didString": "did:cheqd:testnet:DAzMqo4MDMxCjgwM",
+            "methodSpecificId": "DAzMqo4MDMxCjgwM",
+            "method": "cheqd"
+    },
+    "didDocument": {
+        "@context": "https://www.w3.org/ns/did/v1",
+        "id": "did:cheqd:testnet:DAzMqo4MDMxCjgwM",
+        "verificationMethod": [{
+            "id": "did:cheqd:testnet:DAzMqo4MDMxCjgwM#key1",
+            "type": "Ed25519VerificationKey2018",
+            "controller": "did:example:123456789abcdefghi",
+            "publicKeyBase58": "H3C2AVvLMv6gmMNam3uVAjZpfkcJCwDwnZn6z3wXmqPV",
+        "authentication": [
+            "did:cheqd:testnet:DAzMqo4MDMxCjgwM#key1"
+        ],
+        "service": [{
+            "id":"did:cheqd:testnet:DAzMqo4MDMxCjgwM#website",
+            "type": "linkedDomains",
+            "serviceEndpoint": "https://cheqd.io/"
+        }]
+    },
+    "didDocumentMetadata": {
+        "created": "2022-07-01T15:34:30Z",
+        "updated": null,
+        "versionId":"57543FA1D9C56033BABBFA3A438E0A149E01BBB89E6D666ACE1243455AA6F2BC",
+        "resources":[{
+            "resourceURI": "did:cheqd:testnet:DAzMqo4MDMxCjgwM/resources/44547089-170b-bcbc-06e46e0089e4",
+            "resourceName": "CommunitySchema",
+            "resourceType": "CL-Schema",
+            "mediaType": "application/json",
+            "created": "2022-07-01T15:34:30Z"
+        }]
+    }   
+}
+```
```

Comment:
@Toktar Is this a real example from our current resolver implementation? If yes, then something is badly broken: you'll see that the `{` and the `[` brackets (and closing equivalents) starting from the Verification Method section onwards are not properly balanced/closed.

If this is just a copy-paste error, then fine it's easily fixed, but otherwise it might indicate that the resolver isn't formatting things correctly (or worse, it's stored incorrectly on ledger).

Reply:
Sorry, it's not my addition https://github.com/cheqd/identity-docs/pull/8/commits/4f77197f0044dcc6cae1a63da8950da43ffcb47d and it's a bit incorrect. Let me change it to a correct one. At least `didDocumentMetadata` has an incorrect format here